components/embedder_traits/lib.rs

```diff
@@ -84,6 +84,10 @@ pub enum EmbedderMsg {
     Alert(String, IpcSender<()>),
     /// Wether or not to follow a link
     AllowNavigation(ServoUrl, IpcSender<bool>),
+    /// Whether or not to allow script to open a new tab/browser
+    AllowOpeningBrowser(IpcSender<bool>),
+    /// A new browser was created by script
+    BrowserCreated(TopLevelBrowsingContextId),
```

Comment:
Question about this new event: when do get `BrowserCreated`? On `window.open` if `AllowOpeningBrowser` returned true? If so, could we make it so that the message `NewBrowser` uses this new message to send back the context id instead of using an IpcSender for that?

Reply:
Seems to work... https://github.com/servo/servo/pull/20678/commits/c5dc92190a8279d0b249e367eac4a10b1a1766d0